### ISSUE TITLE
contentsScaleFactor can only have one value on multi-window app on Windows 10

### ISSUE THREAD

**Issue by JanFW** [airsdk/Adobe-Runtime-Support#309]
[From [https://tracker.adobe.com/#/view/AIR-4198669](https://tracker.adobe.com/#/view/AIR-4198669) as a reminder that this would be great to have fixed]

**Problem Description:**
stage.contentsScaleFactor can only have one value on multi-window app on Windows 10, causing the app to auto scale when moving or creating windows to other monitors.

**Please refer to:**
https://forums.adobe.com/thread/2488198 (here is an example/demonstration)
https://forums.adobe.com/thread/2498975

**Steps to Reproduce:**
"requestedDisplayResolution" in application descriptor xml needs to be set to "high"

Monitor 1: 1920 x 1080px, 150% Scale
Monitor 2: 1920 x 1080px, 100% Scale

Window 1 on Monitor 1 initially reports stage.contentsScaleFactor = 1.5, which is correct.

Window 2 is created on Monitor 2, reporting stage.contentsScaleFactor = 1, which is also correct - however, Window 1 now also reports contentsScaleFactor = 1, which is not correct.

**Actual Result:**
contentsScaleFactor = 1 for Window 1 on Monitor 1
contentsScaleFactor = 1 for Window 2 on Monitor 2

**Expected Result:**
contentsScaleFactor = 1.5 for Window 1 on Monitor 1
contentsScaleFactor = 1 for Window 2 on Monitor 2

**Any Workarounds:**
None

[src.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4543985/src.zip)

**Comment by JanFW** [comment]
[src.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4543983/src.zip)

**Comment by ajwfrost** [comment]
Should be fixed in 33.1.1.533, please check and re-open if necessary.
Thanks

**Comment by 2jfw** [comment]
Thanks!
I think now the time will come where our team will switch AIR SDK from Adobe Free to Harman Enterprise Tier

**Comment by 2jfw** [comment]
_[It looks like I cannot reopen this issue]_

@ajwfrost: I finally was able to test this (using AIR SDK 33.1.1.575) and found out that when opening a second window, the initial window on the 4k monitor will change its scaling to the scaling of the window which just got opened. Moving the initial window to the other monitor (FullHD) and back to the 4k will correctly scale the initial window.

Overall, the behavior was improved but is still lacking this detail in order to have this issue resolved.

**Comment by 2jfw** [comment]
@ajwfrost Any chance someone may take a look into this? I feel that there is only this one minor issue (see comment above) left in order to have it work

**Comment by ajwfrost** [comment]
@2jfw yes I just checked on that and we have a fix that has just been submitted, so this should work in better in our next release. Thanks for the reminder!

**Comment by 2jfw** [comment]
@ajwfrost Great News! Thanks Andrew! Will report back here when the next release is out asap

**Comment by 2jfw** [comment]
@ajwfrost: The issue is fixed with **AIR 33.1.1.633** - thank you very much 👍 I think its now time for us to move from Adobe AIR to Harman Air finally :)

**Comment by 2jfw** [comment]
[@ajwfrost @marchbold please close this ticket]

**Comment by ajwfrost** [comment]
Great :-) thanks for confirming

**Comment by 2jfw** [comment]
@ajwfrost : Could you please let me know, if there is any _Event_ dispatched by the stage, when the _contentsScaleFactor_ changes?
If not, do you have any recoomendations on how to listen to a change (e.g. _ChangeWatcher / BindingUtils")?

**Comment by daghendrik** [comment]
I'm experimenting with AIR 33.1.1.633 (30-Sep-2021) and a multi-window Flex 4.16 application on Windows 10. I have two monitors connected with different display scaling (125% on primary, 150 on secondary). 

I am really happy that AIR release 33.1.1.633 fixed this bug (#309) with the the contentScaleFactor issue, but I now I've encountered another problem: 

The flash.display.Screen class reports correct (unscaled) bounds for both my primary screen and secondary screen. The X, Y, WIDTH & HEIGHT values are correct and the same as reported by Windows when using  `<requestedDisplayResolution>high</requestedDisplayResolution>` in the Application Descriptor File. All is well so far.
However, if I try to programmatically move a window (for example from the primary to the secondary screen), using the values reported by the Screen class (so that I can make the window cover the whole screen), both the positioning and scaling of the moved window gets messed up.

A workaround (that at least works for my current monitor & scaling setup) is to first move the window to the desired screen, triggering a change of the contentScaleFactor of the moved window's nativeWindow.stage, and then use that "fresh" contentScaleFactor to resize the window to fit the new screen by dividing the screen's bounds with the contentScaleFactor, like this:

1.  First move the s:Window to the secondary screen

```
private function moveToSecondaryScreen():void {
    trace("contentScaleFactor of HideWindow BEFORE move is: " + win.nativeWindow.stage.contentsScaleFactor);
    this.win.nativeWindow.x = secondaryScreen.bounds.x;
    this.win.nativeWindow.y = secondaryScreen.bounds.y;
    this.win.nativeWindow.width = secondaryScreen.bounds.width;
    this.win.nativeWindow.height = secondaryScreen.bounds.height;
}
```

2.  Then resize, based on new contentScaleFactor

```
private function resizeToFitSecondaryScreen():void {
    trace("contentScaleFactor of HideWindow AFTER move is: " + win.nativeWindow.stage.contentsScaleFactor);
    this.win.nativeWindow.x = secondaryScreen.bounds.x / this.win.stage.contentsScaleFactor ;
    this.win.nativeWindow.y = secondaryScreen.bounds.y / this.win.stage.contentsScaleFactor ;
    this.win.nativeWindow.width = secondaryScreen.bounds.width / this.win.stage.contentsScaleFactor ;
    this.win.nativeWindow.height = secondaryScreen.bounds.height / this.win.stage.contentsScaleFactor;
}
```

In my setup, with a secondary screen of 1920x1080 with 150% scaling, the screen class will report the correct bounds, while the window class (after it's resized to fit, using the function above) will report the "scaled" bounds:

**flash.display.Screen**
- x: -1920
- y: 0
- width: 1920
- height: 1080

**Window.nativeWindow**
- x: -1280
- y: 0
- width: 1280
- height: 720

I'd rather avoid this workaround (that I'm not even sure will work for all scaling & monitor configurations) and just programmatically move a window to the coordinates and bounds reported by the flash.display.Screen and have it fit correctly.

Not sure if this can be classified as a bug, but hoping it can be improved in a later release. 

Demo code available at https://github.com/daghendrik/HarmanAir331ScreenTest

I'm happy to open up a separate bug issue for this, if you think it classifies as that @ajwfrost
